service/src/main/java/org/jboss/sbomer/service/feature/sbom/service/AdvisoryService.java

```diff
@@ -303,20 +305,23 @@ protected Collection<SbomGenerationRequest> processDockerBuilds(
         Collection<SbomGenerationRequest> sbomRequests = new ArrayList<SbomGenerationRequest>();
         SyftImageConfig config = SyftImageConfig.builder().withIncludeRpms(true).build();
 
-        buildDetails.forEach((pVersion, items) -> {
-            log.debug("Processing container builds of Errata Product Version: '{}'", pVersion);
-
-            items.forEach(item -> {
-                log.debug("Getting image information from Brew for build '{}' ({})", item.getNvr(), item.getId());
-
-                String imageName = getImageNameFromBuild(item.getId());
-                if (imageName != null) {
-                    config.setImage(imageName);
-                    log.debug("Creating GenerationRequest Kubernetes resource...");
-                    sbomRequests.add(sbomService.generateSyftImage(requestEvent, config));
-                }
-            });
+        // Collect all the docker build ids so we can query Koji in one go
+        List<Long> buildIds = buildDetails.values()
+                .stream()
+                .flatMap(List::stream)
+                .map(BuildItem::getId)
+                .collect(Collectors.toList());
+
+        Map<Long, String> imageNamesFromBuilds = getImageNamesFromBuilds(buildIds);
+        imageNamesFromBuilds.forEach((buildId, imageName) -> {
+            log.debug("Retrieved imageName '{}' for buildId {}", imageName, buildId);
+            if (imageName != null) {
+                config.setImage(imageName);
```

Comment:
Shouldn't we create in the loop the `config`  object? My worry is that if we reuse the object, we may interfere with the persistence layer and (maybe) accidentally sync the wrong thing to the database when the `SbomGenerationRequest.sync` is called.

Reply:
Mmmm ok, maybe. I moved it in the inner loop. Thanks!